app/src/test/java/org/mifos/selfserviceapp/BeneficiaryApplicationPresenterTest.java

```diff
@@ -0,0 +1,153 @@
+package org.mifos.selfserviceapp;
+
+import android.content.Context;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mifos.selfserviceapp.api.DataManager;
+import org.mifos.selfserviceapp.models.beneficary.BeneficiaryPayload;
+import org.mifos.selfserviceapp.models.beneficary.BeneficiaryUpdatePayload;
+import org.mifos.selfserviceapp.models.templates.beneficiary.BeneficiaryTemplate;
+import org.mifos.selfserviceapp.presenters.BeneficiaryApplicationPresenter;
+import org.mifos.selfserviceapp.ui.views.BeneficiaryApplicationView;
+import org.mifos.selfserviceapp.util.RxSchedulersOverrideRule;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import okhttp3.ResponseBody;
+import rx.Observable;
+
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+/**
+ * Created by dilpreet on 12/7/17.
+ */
+@RunWith(MockitoJUnitRunner.class)
+public class BeneficiaryApplicationPresenterTest {
+
+    @Rule
+    public final RxSchedulersOverrideRule mOverrideSchedulersRule = new RxSchedulersOverrideRule();
+
+    @Mock
+    Context context;
+
+    @Mock
+    DataManager dataManager;
+
+    @Mock
+    BeneficiaryApplicationView view;
+
+    @Mock
+    ResponseBody mockedResponseBody;
+
+    private BeneficiaryUpdatePayload updatePayload;
+    private BeneficiaryPayload payload;
+    private BeneficiaryApplicationPresenter presenter;
+    private BeneficiaryTemplate beneficiaryTemplate;
+
+    @Before
+    public void setUp() {
+        presenter = new BeneficiaryApplicationPresenter(dataManager, context);
+        presenter.attachView(view);
+
+        beneficiaryTemplate = FakeRemoteDataSource.getBeneficiaryTemplate();
+        payload = FakeRemoteDataSource.beneficiaryPayload();
+        updatePayload = FakeRemoteDataSource.beneficiaryUpdatePayload();
+
+        when(context.getString(R.string.error_fetching_beneficiary_template)).
```

Comment:
@therajanmaurya as these are junit tests cases and I am trying to mock `getString()` method so how can we use strings.xml values in `thenReturn()` function?